kie-server-parent/kie-server-controller/kie-server-controller-rest/src/main/java/org/kie/server/controller/rest/RestRuntimeManagementServiceImpl.java

```diff
@@ -0,0 +1,110 @@
+
+            return createCorrectVariant(response, headers, Response.Status.OK);
+        } catch (KieServerControllerIllegalArgumentException e) {
+            return createCorrectVariant(e.getMessage(), headers, Response.Status.NOT_FOUND);
+        } catch (KieServerControllerException e){
+            return createCorrectVariant(REQUEST_FAILED_TOBE_PROCESSED + e.getMessage(), headers, Response.Status.BAD_REQUEST);
+        } catch (Exception e) {
+            logger.error("Get server instances using server template id {} failed due to {}", serverTemplateId, e.getMessage(), e);
+            return createCorrectVariant("Unknown error " + e.getMessage(), headers, Response.Status.INTERNAL_SERVER_ERROR);
+        }
+
+    }
+
+    @GET
+    @Path("servers/{id}/instances/{instanceId}/containers")
+    @Produces({MediaType.APPLICATION_XML, MediaType.APPLICATION_JSON})
+    public Response getServerInstanceContainers(@Context HttpHeaders headers, @PathParam("id") String serverTemplateId, @PathParam("instanceId") String instanceId) {
+        String contentType = getContentType(headers);
+        try {
+            logger.debug("Received get container {} for server template with id {} and instance id {}", serverTemplateId, instanceId);
+
+            ContainerList containers = super.getServerInstanceContainers(serverTemplateId, instanceId);
+            String response = marshal(contentType, containers);
+            logger.debug("Returning response for get containers for server template with id {} and instance id {}: {}", serverTemplateId, instanceId, response);
+
+            return createCorrectVariant(response, headers, Response.Status.OK);
+        } catch (KieServerControllerIllegalArgumentException e) {
+            return createCorrectVariant(e.getMessage(), headers, Response.Status.NOT_FOUND);
+        } catch (KieServerControllerException e){
+            return createCorrectVariant(REQUEST_FAILED_TOBE_PROCESSED + e.getMessage(), headers, Response.Status.BAD_REQUEST);
+        } catch (Exception e) {
+            logger.error("Get containers using server template id {} and instance id {} failed due to {}", serverTemplateId, instanceId, e.getMessage(), e);
+            return createCorrectVariant("Unknown error " + e.getMessage(), headers, Response.Status.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    @GET
+    @Path("servers/{id}/containers/{containerId}")
+    @Produces({MediaType.APPLICATION_XML, MediaType.APPLICATION_JSON})
+    public Response getServerTemplateContainers(@Context HttpHeaders headers, @PathParam("id") String serverTemplateId, @PathParam("containerId") String containerId) {
+        String contentType = getContentType(headers);
+        try {
+            logger.debug("Received get container {} for server template with id {}", containerId, serverTemplateId);
+
+            ContainerList containers = super.getServerTemplateContainers(serverTemplateId, containerId);
```

Comment:
why this returns a list of containers? shouldn't there be only one container with given id in server template?

Reply:
I'm not quite sure, I thought this would be the list of containers with the same id in all server instances related to a server template. It really just replicates/exposes the following method: https://github.com/kiegroup/droolsjbpm-integration/blob/master/kie-server-parent/kie-server-controller/kie-server-controller-api/src/main/java/org/kie/server/controller/api/service/RuntimeManagementService.java#L28

Reply:
in that case the endpoint should be renamed to something that would make it clearer like
servers/{id}/containers/{containerId}/instances
as this essentially returns list of actual instances running, isn't it?

Reply:
@mswiderski I was considering using _servers/{id}/instances/containers/{containerId}_ to keep consistency with the other endpoints. The difference here is that there is no filter regarding a particular server instance. wdty?

Reply:
the problem I see with it is that ending it with {containerId} indicates there will be single resource returned while here we get a list so having this end with /instances makes more sense to me at least.